packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -182,6 +155,107 @@ export class GitDiffWidget extends GitBaseWidget implements StatefulWidget {
+                }
+            });
+        } else if (this.gitNodes.length > 0) {
+            this.selectNode(this.gitNodes[0]);
+            this.openSelected();
+        }
+    }
+
+    protected handleLeft(): void {
+        const selected = this.getSelected();
+        if (GitFileChangeNode.is(selected)) {
+            const uri = this.getUriToOpen(selected);
+            this.editorManager.getByUri(uri).then(widget => {
+                if (widget) {
+                    const diffNavigator: DiffNavigator = this.diffNavigatorProvider(widget.editor);
+                    if (diffNavigator.canNavigate() && diffNavigator.hasPrevious()) {
+                        diffNavigator.previous();
+                    } else {
+                        this.selectPreviousNode();
+                        this.openSelected();
+                    }
+                } else {
+                    this.openFile(selected);
+                }
+            });
+        }
+    }
+
+    protected handleEnter(): void {
+        this.openSelected();
+    }
+
+    protected openSelected(): void {
+        const selected = this.getSelected();
+        if (selected) {
+            this.openFile(selected);
+        }
+    }
+
+    protected getUriToOpen(change: GitFileChange): URI {
+        const uri: URI = new URI(change.uri);
+
+        let fromURI = uri;
+        if (change.oldUri) { // set on renamed and copied
+            fromURI = new URI(change.oldUri);
+        }
+        if (this.fromRevision !== undefined) {
+            if (typeof this.fromRevision !== 'number') {
+                fromURI = fromURI.withScheme(GIT_RESOURCE_SCHEME).withQuery(this.fromRevision);
+            } else {
+                fromURI = fromURI.withScheme(GIT_RESOURCE_SCHEME).withQuery(this.toRevision + "~" + this.fromRevision);
+            }
+        } else {
+            // default is to compare with previous revision
+            fromURI = fromURI.withScheme(GIT_RESOURCE_SCHEME).withQuery(this.toRevision + "~1");
+        }
+
+        let toURI = uri;
+        if (this.toRevision) {
+            toURI = toURI.withScheme(GIT_RESOURCE_SCHEME).withQuery(this.toRevision);
+        }
+
+        let uriToOpen = uri;
+        if (change.status === GitFileStatus.Deleted) {
+            uriToOpen = fromURI;
+        } else if (change.status === GitFileStatus.New) {
+            uriToOpen = toURI;
+        } else {
+            uriToOpen = DiffUris.encode(fromURI, toURI, uri.displayName);
+        }
+        return uriToOpen;
+    }
+
+    protected openFile(change: GitFileChange) {
+        const uriToOpen = this.getUriToOpen(change);
+        open(this.openerService, uriToOpen, { mode: 'reveal' }).
+            then(widget => {
+                if (widget instanceof EditorWidget) {
+                    const diffNavigator: DiffNavigator = this.diffNavigatorProvider(widget.editor);
+                    if (diffNavigator.canNavigate() && diffNavigator.hasNext()) {
```

Comment:
I'd expect, that the diff editor reveals first change with #1255 already. OTOH it would make sense to start over with the selection, if an existing diff editor would be just revealed.